src/executors/mod.rs

```diff
@@ -105,12 +106,16 @@ pub async fn execute_middleware_function<'a>(
                 // is this a bottleneck again?
                 request.insert("headers", headers.clone().into_py(py));
                 request.insert("body", data.into_py(py));
+                response_dict.insert("headers", response_headers.into_py(py));
+                response_dict.insert("status", response_status_code.into_py(py));
+                response_dict.insert("body", response_body.into_py(py));
 
                 let output: PyResult<&PyAny> = match number_of_params {
                     0 => handler.call0(),
                     1 => handler.call1((request,)),
+                    2 => handler.call1((request, response_dict)),
                     // this is done to accomodate any future params
-                    2_u8..=u8::MAX => handler.call1((request,)),
+                    3_u8..=u8::MAX => handler.call1((request, response_dict)),
```

Comment:
Everything looks great. Just one last nit:

`request` and `response_dict` do not match. Can you rename them to `request` and `response` or  `req` and `res` or `request_dict` and `response_dict`?